src/register/EmbeddableRegistrationPage.jsx

```diff
@@ -303,16 +374,26 @@ const EmbeddableRegistrationPage = (props) => {
     Object.keys(configurableFormFields).forEach((fieldName) => {
       if (fieldName === 'country') {
         payload[fieldName] = configurableFormFields[fieldName].countryCode;
+        validatedFormFields[fieldName] = configurableFormFields[fieldName].countryCode;
       } else {
         payload[fieldName] = configurableFormFields[fieldName];
       }
     });
-
     // Don't send the marketing email opt-in value if the flag is turned off
     if (!flags.showMarketingEmailOptInCheckbox) {
       delete payload.marketingEmailsOptIn;
     }
-
+    let isDirty = false;
+    const dirtyFormFields = {};
+    Object.entries(validatedFormFields).forEach(([key, value]) => {
+      ({ isDirty } = validateInput(key, value, payload, false, true));
+      if (isDirty) {
```

Comment:
this check can be applied on fieldError returned by validateInput (as per my comment in validateInput function return)